examples/crowdfunding/test/crowdfundingTest.js

```diff
@@ -0,0 +1,252 @@
+    assert.deepEqual(runtime.getGlobalState(appId, stringToBytes('Total')), BigInt(0));
+  });
+
+  it('should update application with correct escrow account address', async () => {
+    const appArgs = [addressToBytes(escrowAccount.address)];
+
+    await runtime.updateApp(
+      creatorAccount.account.addr,
+      applicationId,
+      program,
+      {}, { appArgs: appArgs });
+    const escrowPk = addressToBytes(escrowAccount.address);
+
+    // verify escrow storage
+    assert.isDefined(applicationId);
+    assert.deepEqual(runtime.getGlobalState(applicationId, stringToBytes('Escrow')), escrowPk);
+  });
+
+  it('should opt-in to app', async () => {
+    await runtime.optInToApp(creatorAccount.address, applicationId, {}, {}, program);
+    await runtime.optInToApp(donorAccount.address, applicationId, {}, {}, program);
+
+    syncAccounts();
+    assert.isDefined(creatorAccount.appsLocalState.find(app => app.id === applicationId));
+    assert.isDefined(donorAccount.appsLocalState.find(app => app.id === applicationId));
+  });
+
+  it('should donate correct amount to escrow account', async () => {
+    // App argument to donate.
+    const appArgs = [stringToBytes('donate')];
+    const donationAmount = 600000;
+
+    // Atomic Transaction (Stateful Smart Contract call + Payment Transaction)
+    const txGroup = [
+      {
+        type: TransactionType.CallNoOpSSC,
+        sign: SignType.SecretKey,
+        fromAccount: donorAccount.account,
+        appId: applicationId,
+        payFlags: {},
+        appArgs: appArgs
+      },
+      {
+        type: TransactionType.TransferAlgo,
+        sign: SignType.SecretKey,
+        fromAccount: donorAccount.account,
+        toAccountAddr: escrowAccount.address,
+        amountMicroAlgos: donationAmount,
+        payFlags: {}
+      }
+    ];
+    // execute transaction
+    await runtime.executeTx(txGroup, program, []);
+
+    syncAccounts();
+    assert.equal(escrowAccount.balance(), donationAmount);
+    assert.equal(donorAccount.balance(), initialDonorBalance - donationAmount);
+  });
+
+  it('should not be able to claim funds because goal is not reached', async () => {
+    const appArgs = [stringToBytes('claim')];
+    // Atomic Transaction (Stateful Smart Contract call + Payment Transaction)
+    const txGroup = [
+      {
+        type: TransactionType.CallNoOpSSC,
+        sign: SignType.SecretKey,
+        fromAccount: creatorAccount.account,
+        appId: applicationId,
+        payFlags: {},
+        appArgs: appArgs
+      },
+      {
+        type: TransactionType.TransferAlgo,
+        sign: SignType.LogicSignature,
+        fromAccount: { addr: escrowAccount.address },
+        toAccountAddr: creatorAccount.address,
+        amountMicroAlgos: 0,
+        lsig: escrowAccount,
+        payFlags: { closeRemainderTo: creatorAccount.address }
+      }
+    ];
+    // execute transaction: Expected to be rejected by logic
```

Comment:
Please add more explanation to this comment (i.e why it's failing)

Reply:
if you read the test description, that is because goal is not reached, will copy paste it here too.